src/intrinsics/ecma262/BooleanPrototype.js

```diff
@@ -10,20 +10,29 @@
 /* @flow */
 
 import type { Realm } from "../../realm.js";
-import { ObjectValue, StringValue } from "../../values/index.js";
+import { ObjectValue, StringValue, AbstractValue, BooleanValue } from "../../values/index.js";
 import { thisBooleanValue } from "../../methods/to.js";
+import { TypesDomain, ValuesDomain } from "../../domains/index.js";
+import buildExpressionTemplate from "../../utils/builder.js";
 
 export default function(realm: Realm, obj: ObjectValue): void {
   // ECMA262 19.3.1
   obj.$BooleanData = realm.intrinsics.false;
 
   // ECMA262 19.3.3.3
   obj.defineNativeMethod("toString", 0, context => {
-    // 1. Let b be ? thisBooleanValue(this value).
-    let b = thisBooleanValue(realm, context);
```

Comment:
Why not unbox if this a boxed Boolean? That way the logic below will work even for boxed abstract Boolean values.